core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -85,9 +94,11 @@ public void prepare(Snapshot snapshot) throws PreparationException {
   }
 
   private void prepareRecords(Snapshot snapshot)
-      throws ExecutionException, PreparationConflictException {
+      throws ExecutionException, PreparationConflictException,
+          PreparationUnsatisfiedConditionException {
     PrepareMutationComposer composer =
         new PrepareMutationComposer(snapshot.getId(), tableMetadataManager);
+    snapshot.validateConditionalMutations();
```

Comment:
Moreover, I'm also wondering if validating conditions should be `Snapshot`'s responsibility or not.

@brfrn169 What do you think?

Reply:
Sorry, I should have elaborated more.
I meant, can we check the conditions of mutation operations when calling the operations, not when committing 
 the transaction? (so that we can do an early abort)
For example, it seems like we can check the conditions of `Put` when calling `CrudHandler.put`.
I might miss something, though.

Reply:
> For example, it seems like we can check the conditions of Put when calling CrudHandler.put.
@feeblefakie Ah, indeed, it seems we could do that.

@Torch3333 What do you think?

Reply:
@brfrn169 
After discussing with @feeblefakie this morning, we concluded it is possible to move the condition check to `CrudHandler.put` and `CrudHandler.delete`.
Regarding the throw exception when the condition is not satisfied, we can then have a single exception type for regular and 2pc transaction interface that is called `UnsatisfiedConditionException` (extending `CrudException`) when executing `transaction.put()` or `transaction.delete()`. 
Overall, it will simplify the implementation.

Reply:
I revised the PR as described above. 
Thank you for the suggestion.